File: src/main/java/io/sch/historyscan/domain/contexts/analysis/history/HistoryAnalyzer.java
package io.sch.historyscan.domain.contexts.analysis.history;

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI;

import java.util.Optional;

@HexagonalArchitectureSPI
@FunctionalInterface
public interface HistoryAnalyzer {
        Optional<CodeBaseHistory> of(String codeBaseName);
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/analysis/history/HistoryAnalyzer.kt
package io.sch.historyscan.domain.contexts.analysis.history

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureSPI

@HexagonalArchitectureSPI
fun interface HistoryAnalyzer {
    fun of(codeBaseName: String): CodeBaseHistory?
}
